front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -346,59 +327,15 @@ async function handler(
       });
 
       const connectorsAPI = new ConnectorsAPI(logger);
-      let connectorsRes: Awaited<ReturnType<ConnectorsAPI["createConnector"]>>;
-      switch (type) {
-        case "oauth": {
-          if (!connectionId) {
-            return apiError(req, res, {
-              status_code: 400,
-              api_error: {
-                type: "invalid_request_error",
-                message: "connectionId is required for OAuth connectors.",
-              },
-            });
-          }
-          connectorsRes = await connectorsAPI.createConnector(
-            provider,
-            owner.sId,
-            systemAPIKeyRes.value.secret,
-            dataSourceName,
-            {
-              connectionId: connectionId,
-            }
-          );
-          break;
-        }
-        case "url": {
-          if (!urlConfig?.url) {
-            return apiError(req, res, {
-              status_code: 400,
-              api_error: {
-                type: "invalid_request_error",
-                message: "url is required for URL connectors.",
-              },
-            });
-          }
-          let cleanUrl = urlConfig.url.trim();
-          if (
-            !cleanUrl.startsWith("http://") &&
-            !cleanUrl.startsWith("https://")
-          ) {
-            cleanUrl = `http://${cleanUrl}`;
-          }
-          connectorsRes = await connectorsAPI.createConnector(
-            provider,
-            owner.sId,
-            systemAPIKeyRes.value.secret,
-            dataSourceName,
-            urlConfig
-          );
-          break;
-        }
 
-        default:
-          assertNever(type);
-      }
+      const connectorsRes = await connectorsAPI.createConnector(
+        provider,
+        owner.sId,
+        systemAPIKeyRes.value.secret,
+        dataSourceName,
+        connectionId || "n/a",
```

Comment:
ConnectionId can be null for webcrawler, we agreed on keeping it as a required field in `connectors.connectionId`, so `n/a` it is. Type is properly checked in the switch case above.

Reply:
I'm in favor of `none` vs `n/a` but that's a total nit